tlair/src/main/scala/at/forsyte/apalache/tla/lir/TlaType1.scala

```diff
@@ -37,6 +39,22 @@ object TlaType1 {
       case _ => throw new TypingException("Expected Typed(_: TlaType1), found: " + typeTag, UID.nullId)
     }
   }
+
+  // returns true iff the type is a monotype, i.e. if it contains on type variables
+  def isMono(tt: TlaType1): Boolean = tt match {
+    case _: VarT1                 => false
+    case FunT1(arg, res)          => isMono(arg) && isMono(res)
+    case SetT1(elem)              => isMono(elem)
+    case SeqT1(elem)              => isMono(elem)
+    case TupT1(elems @ _*)        => elems.forall(isMono)
+    case SparseTupT1(fieldTypes)  => fieldTypes.values.forall(isMono)
+    case RecT1(fieldTypes)        => fieldTypes.values.forall(isMono)
+    case OperT1(args, res)        => (res +: args).forall(isMono)
+    case RowT1(fieldTypes, other) => (other ++ fieldTypes.values).forall(isMono)
+    case RecRowT1(row)            => isMono(row)
+    case VariantT1(row)           => isMono(row)
+    case _                        => true
```

Comment:
This will end up leading to false positives anytime a new constructor is added. Let's not have a wildcard match here, and instead require enumeration of all options?

```suggestion
```

Reply:
How much of an issue is this really? We're not adding more types anytime soon, and besides, 95% of our matches have a default branch, because it avoids code duplication.

Reply:
This is a bug waiting to happen, and it will guarantee incorrect behavior, unless the person making the change explicitly writes a test for it. Meanwhile, iiuc, it should be trivially easy to ensure all the cases are enumerated, and statically guarantee this bug wont arise. So why not do it?